android/cli/commands/_build.js

```diff
@@ -1472,24 +1472,32 @@ AndroidBuilder.prototype.validate = function validate(logger, config, cli) {
 				if (module.platform.indexOf('commonjs') != -1) {
 					module.native = false;
 
-					// Look for legacy module.id.js first
-					module.libFile = path.join(module.modulePath, module.id + '.js');
-					if (!fs.existsSync(module.libFile)) {
-						// then package.json TODO Verify the main property points at reale file under the module!
-						module.libFile = path.join(module.modulePath, 'package.json');
-						if (!fs.existsSync(module.libFile)) {
-							// then index.js
-							module.libFile = path.join(module.modulePath, 'index.js');
-							if (!fs.existsSync(module.libFile)) {
-								// then index.json
-								module.libFile = path.join(module.modulePath, 'index.json');
-								if (!fs.existsSync(module.libFile)) {
-									this.logger.error(__('Module %s version %s is missing module files: %s, package.json, index.js, or index.json', module.id.cyan, (module.manifest.version || 'latest').cyan, path.join(module.modulePath, module.id + '.js').cyan) + '\n');
-									process.exit(1);
+					// look for legacy module.id.js first
+					var libFile = path.join(module.modulePath, module.id + '.js');
+					module.libFile = fs.existsSync(libFile) ? libFile : null;
+					if (!module.libFile) {
+						var pkgJsonFile = path.join(module.modulePath, 'package.json');
+						if (fs.existsSync(pkgJsonFile)) {
+							try {
+								var pkgJson = require(pkgJsonFile);
+								if (pkgJson && pkgJson.main && fs.existsSync(libFile = path.join(module.modulePath, pkgJson.main + (/\.js$/.test(pkgJson.main) ? '' : '.js')))) {
```

Comment:
I updated the code as it existed. I was unaware that libFile was unused, but I viewed this block as basic validation of the structure of CommonJS modules. As such, I updated the logic to check for more than the module_id.js file so npm style modules would work.

So yeah, we could nuke the block and just assume that the CommonJS modules are ok, and bomb at runtime instead if they're not. Or we could retain the validation and improve it.